src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -75,6 +75,8 @@ protected function createController($controller)
             return $service;
         }
 
+        $this->throwExceptionIfControllerWasRemoved($controller);
```

Comment:
L70 can also fail for the same reason

Reply:
Yes, I think it's a good idea to add a similar check there too. Currently you would get this error:

> The "app.my_controller" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.

Reply:
But that's fine, it fails with the following exception:
> The "app.my_controller" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.

Reply:
It feels inconsistent to me to get different errors depending on how you configured the route.

Reply:
I agree with @xabbuh :)

Reply:
Refactored it and added another DX improvement for missing `__invoke` methods at the same time 😉